pages/employee/index.vue

```diff
@@ -0,0 +1,142 @@
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup name="Employment Type" label="Employment Type">
+                    <UInputMenu v-model="employee.employe_type" :options="employeeTypes" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup :label="incomeTitle" name="income">
+                    <UInput :placeholder="`Enter ${incomeTitle}`" v-model="employee.income" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UButton type="submit">
+                    Submit
+                </UButton>
+            </div>
+        </div>
+    </UForm>
+</template>
+
+<script setup lang="ts">
+import type { FormError, FormErrorEvent, FormSubmitEvent } from '#ui/types'
+const employeeTypes = ['Business', 'Salaried']
+const employee = ref({
+    first_name: undefined,
+    last_name: undefined,
+    mobile: undefined,
+    pincode: undefined,
+    employe_type: employeeTypes[0],
+    income: undefined,
+})
+let pincodeError = ref('')
+let formChanged = ref(false)
+const allowNumericValue = (event: InputEvent) => {
+    const input = event.target as HTMLInputElement;
+    input.value = input.value.replace(/\D/g, '');
+    formChanged.value = true;
+};
+const validate = (employee: any): FormError[] => {
+    const errors = []
+    if (!employee.first_name) errors.push({ path: 'first_name', message: 'First Name Required' })
+    if (!employee.last_name) errors.push({ path: 'last_name', message: 'Last Name Required' })
+
+    if (!employee.employe_type) errors.push({ path: 'employe_type', message: 'Employee Type Required' })
+    if (!employee.mobile) {
+        errors.push({ path: 'mobile', message: 'Mobile Number Required' })
+    } else if (!/^\d{10}$/.test(employee.mobile)) {
+        errors.push({ path: 'mobile', message: 'Mobile Number should be  10 digits ' });
+    }
+    if (!employee.pincode) {
+        errors.push({ path: 'pincode', message: 'Pincode Required' })
+    } else if (!/^\d{6}$/.test(employee.pincode)) {
+        errors.push({ path: 'pincode', message: 'Pincode should be  6 digits ' });
+    }
+    formChanged.value = true;
+
+    return errors
+}
+
+const incomeTitle = computed(() => {
+    return employee.value.employe_type === 'Business' ? 'Income' : 'Salary';
+});
+function submitEmployee() {
+    if (pincodeError.value === '') {
+        sessionStorage.setItem('employee', JSON.stringify(employee.value));
+        formChanged.value = false;
+    }
+}
+
+async function onError(event: FormErrorEvent) {
+    const element = document.getElementById(event.errors[0].id)
+    element?.focus()
+    element?.scrollIntoView({ behavior: 'smooth', block: 'center' })
+}
+async function checkPincode() {
+    try {
+        const response = await fetch(`https://api.postalpincode.in/pincode/${employee.value.pincode}`)
```

Comment:
we can fetch city / state  and show them below the pincode input

Reply:
Resolved